osquery/filesystem/filesystem.cpp

```diff
@@ -266,15 +277,33 @@ Status resolveLastPathComponent(const boost::filesystem::path& fs_path,
   }
 
   std::vector<std::string> files;
-  Status stat = listFilesInDirectory(fs_path.parent_path(), files);
-  if (!stat.ok()) {
-    return stat;
+  std::vector<std::string> folders;
+  Status stat_file = listFilesInDirectory(fs_path.parent_path(), files);
+  Status stat_fold = listDirectoriesInDirectory(fs_path.parent_path(), folders);
+
+  if (!stat_file.ok()) {
+    return stat_file;
+  }
+  if (!stat_fold.ok()) {
+    return stat_fold;
   }
 
   // Is the last component a wildcard?
   if (components[components.size() - 1] == kWildcardCharacter) {
-    for (const auto& file : files) {
-      results.push_back(file);
+    if (setting & REC_EVENT_OPT) {
+      results.push_back(fs_path.parent_path().string());
+      return Status(0, "OK");
+    }
+    if (setting & REC_LIST_FOLDERS) {
+      results.push_back(fs_path.parent_path().string());
```

Comment:
No because if we add it with the REC_EVENT_OPT flag we return and cannot add it again.